app/grandchallenge/algorithms/migrations/0085_algorithm_social_image_height_and_more.py

```diff
@@ -0,0 +1,91 @@
+# Generated by Django 4.2.26 on 2025-11-13 09:08
+
+import pictures.models
+from django.db import migrations, models
+from pictures.migrations import AlterPictureField
+
+import grandchallenge.core.storage
+
+
+def add_social_image_dimensions(apps, schema_editor):
+    for obj in apps.get_model("algorithms.algorithm").objects.all().iterator():
+        if obj.social_image:
+            obj.social_image_width = obj.social_image.width
+            obj.social_image_height = obj.social_image.height
+        obj.save(update_fields=["social_image_width", "social_image_height"])
+
+
+def remove_social_image_dimensions(apps, schema_editor):
+    apps.get_model("algorithms.algorithm").objects.all().update(
+        social_image_width=None,
+        social_image_height=None,
+    )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("algorithms", "0084_job_exec_duration_job_invoke_duration"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="algorithm",
+            name="social_image_height",
+            field=models.PositiveSmallIntegerField(editable=False, null=True),
+        ),
+        migrations.AddField(
+            model_name="algorithm",
+            name="social_image_width",
+            field=models.PositiveSmallIntegerField(editable=False, null=True),
+        ),
+        migrations.RunPython(
```

Comment:
We usually do these separately, but I am trying to see if this works altogether in production now. Will need the staging environment for this.